src/main/resources/META-INF/resources/primefaces-extensions/fuzzysort/primefaces.fuzzysort.js

```diff
@@ -0,0 +1,52 @@
+PrimeFaces.widget.Fuzzysort = PrimeFaces.widget.BaseWidget.extend({
+
+    init: function (cfg) {
+        this.cfg = cfg;
+        this.id = this.cfg.id;
+        this.jqId = PrimeFaces.escapeClientId(this.id);
+//        this.jq = $(PrimeFaces.escapeClientId(this.cfg.target));
+
+        this.input = $(PrimeFaces.escapeClientId(this.cfg.id + "_fuzzysort-search-box"));
+        this.results = $(PrimeFaces.escapeClientId(this.cfg.id + "_fuzzysort-search-results"));
+        this.datasource = JSON.parse(this.cfg.value);
+
+        var $this = this;
+//        console.log($this);
+
+        function search() { // TODO how to move search function to search_alternative
+            $this.results.empty();
```

Comment:
You should be able to move this code into search_alternative just change the $this to this when you move it.